File: src/main/java/ParseInput.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.InputStreamReader;

public class ParseInput {

    static int processCount;
    static int processId[];
    static int rootProcess;
    static int adjacencyMatrix[][];

    public static void readFile(String fileName) {
        File file = new File(fileName);
        if (!file.exists()) {
            Log.write("ParseInput", "Input File not found");
        } else {
            try {
                BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(file)));
                String line;
                int count = 0;
                while ((line = br.readLine()) != null) {
                    if (count == 0)
                        processCount = Integer.parseInt(line);
                    else if (count == 1) {
                        processId = new int[processCount];
                        String ids[] = line.split(" ");
                        for (int i = 0; i < ids.length; i++)
                            processId[i] = Integer.parseInt(ids[i]);
                    } else if (count == 2)
                        rootProcess = Integer.parseInt(line);
                    else {
                        if (count == 3)
                            adjacencyMatrix = new int[processCount][processCount];
                        String ids[] = line.split(" ");
                        for (int i = 0; i < ids.length; i++)
                            adjacencyMatrix[count - 3][i] = Integer.parseInt(ids[i]);
                    }
                    count += 1;
                }
            } catch (Exception e) {
                System.out.println(e.getMessage());
            }
        }

    }
}

File: src/ParseInput.kt
import java.io.BufferedReader
import java.io.File
import java.io.FileInputStream
import java.io.InputStreamReader

object ParseInput {

    var processCount: Int = 0
    var processId: IntArray = IntArray(1)
    var rootProcess: Int = 0
    var adjacencyMatrix: Array<IntArray> = Array(0) { IntArray(0) }

    fun readFile(file: String) {
        val f = File(file)
        if (!f.exists())
            Log.write("ParseInput", "Input File not found")
        else {
            val br = BufferedReader(InputStreamReader(FileInputStream(f)))
            var count = 0
            var line = br.readLine()
            while (line != null) {
                when (count) {
                    0 -> processCount = line.toInt()
                    1 -> {
                        processId = IntArray(processCount)
                        val ids = line.split(" ")
                        for (i in 0..(processCount-1)) processId[i] = ids[i].toInt()
                    }
                    2 -> rootProcess = line.toInt()
                    else -> {
                        if (count == 3) adjacencyMatrix = Array(processCount) { IntArray(processCount) }
                        val ids = line.split(" ")
                        for (i in 0..(processCount-1)) adjacencyMatrix[count - 3][i] = ids[i].toInt()
                    }
                }
                count++
                line = br.readLine()
            }
        }
    }

}